### ISSUE TITLE
Can't manage to get an element to scroll

### ISSUE THREAD

**Issue by andy-dorman** [wix/Detox#2367]
I'm not entirely sure I need to post much more than this but can anyone tell me why I'm unable to scroll an element that I've confirmed is visible? It fails in both Android and IOS, using either accessibilityLabels or testIDs. The element in question is a React Native ScrollView and all I want to do is scroll a button at the bottom of it into view. I know I can use whileElement and I'd prefer to use testIDs, I'll get to that but after an afternoon of going back and forth this is the simplest failing example I can give.

Detox looks really great and we'd love to use it but I'm decidedly unimpressed at the moment that I can't implement this simplest of test cases.

Many, many thanks in advance - I can only assume this is user error...

```
    Test Failed: No views in hierarchy found matching: (with content description text: is "ENVIRONMENT_SELECTION_PANEL" and view has effective visibility=VISIBLE)

    TIP: To print view hierarchy on failed actions/matches, use loglevel verbose and above.

      19 |         /* END */
      20 |         await waitFor(element(by.label('ENVIRONMENT_SELECTION_PANEL'))).toBeVisible()
    > 21 |         await element(by.label('ENVIRONMENT_SELECTION_PANEL')).scroll(200, 'down')
         |                                                                ^
      22 |         await element(by.text('Done')).tap()
      23 |     })
      24 | })
```

**Comment by andy-dorman** [comment]
If you say so, but in the absence of any other information it kind of looks like a bug to me but ok...

**Comment by LeoNatan** [comment]
> Test Failed: No views in hierarchy found matching: (with content description text: is "ENVIRONMENT_SELECTION_PANEL" and view has effective visibility=VISIBLE)

Kinda tells the tale.

**Comment by andy-dorman** [comment]
I'll choose to ignore that as I assume the majority of people looking at detox have some level of professional experience.

It turns out that this breaks it:

```
<ScrollView
  style={{
    flex: 1
  }}
  contentContainerStyle={{
        flex: 1,
        justifyContent: 'space-between'
    }}
>
```
whereas this works as expected:
```
<ScrollView
  style={{
    flex: 1
  }}
  contentContainerStyle={{
        flexGrow: 1,
        justifyContent: 'space-between'
    }}
>
```
as does this:
```
<ScrollView
  style={{
    flex: 1
  }}
>
```
So actually it's an issue with React Native's handling of the flex css property as documented here:

https://medium.com/@peterpme/taming-react-natives-scrollview-with-flex-144e6ff76c08

I would suggest that it may be valuable to see if you can detect these kind of cases and display an appropriate error message as the current one is misleading and unhelpful. Particularly as I get it the test to pass/fail only by adjusting the markup leaving the test itself looking like this:
```
await waitFor(element(by.id('panel')))
  .toBeVisible()
  .withTimeout(1000)
await expect(element(by.id('panel'))).toBeVisible()
await element(by.id('panel')).scrollTo('bottom')
```
With the only difference being whether that final action succeeds or fails.

I'll add this to the stack overflow post here:

https://stackoverflow.com/questions/64116310/how-to-scroll-elements-in-detox

in case any other wandering travellers experience the same issue in the future.
